src/plugins/intel_gpu/src/kernel_selector/cl_kernels/softmax_gpu_bf.cl

```diff
@@ -4,43 +4,70 @@
 
 #include "include/batch_headers/common.cl"
 
+#if IS_DYNAMIC
+#define CALC_POWER(n) ({uint pos = 0; uint i = n; do { i >>= 1; ++pos; } while (i); --pos;})
+#endif
+
+#if !IS_DYNAMIC
 __attribute__((reqd_work_group_size(LWS, 1, 1)))
+#endif
 KERNEL (softmax_gpu_continuous_bfyx)(
+    OPTIONAL_SHAPE_INFO_ARG
     const __global INPUT0_TYPE* input,
+#ifdef IS_DYNAMIC
+    __global INPUT0_TYPE* tmp_buffer,
+#endif
     __global OUTPUT_TYPE* output
 #if HAS_FUSED_OPS_DECLS
     , FUSED_OPS_DECLS
 #endif
 ) {
-    const uint data_set_idx = get_global_id(1);     //in processing of which data set this WI participates?
-    const uint workers_per_data_set = LWS;          //how many WI participates in processing of one data set
-    const uint in_data_set_idx = get_global_id(0);  //this WI's id in group of items processing single data set
-    const uint data_set_size = DATA_SET_SIZE;       //how many elements are in one data set
-    const uint data_sets_count = DATA_SETS_COUNT;   //how many data sets are in the processing payload
+    const uint data_set_idx = get_global_id(1);     // in processing of which data set this WI participates?
+    const uint workers_per_data_set = LWS;          // how many WI participates in processing of one data set
+    const uint in_data_set_idx = get_global_id(0);  // this WI's id in group of items processing single data set
+    const uint data_set_size = DATA_SET_SIZE;       // how many elements are in one data set
+    const uint data_sets_count = DATA_SETS_COUNT;   // how many data sets are in the processing payload
+#if !IS_DYNAMIC
+    const uint items_num = ITEMS_NUM;               // how many elements are processed per one WI
+    const uint leftovers = LEFTOVERS;
+#else
+    // since workers_per_data_set is calculated by power of 2
+    // items_num can be calculated by dividing data_set_size by power of 2
+    const uint power = CALC_POWER(workers_per_data_set);
+    const uint items_num = data_set_size>>power;
+    const uint leftovers = data_set_size-(items_num<<power);
+#endif
 
     const uint data_set_offset = data_set_idx * data_set_size;
     const uint my_data_offset = data_set_offset + in_data_set_idx;
 
-    INPUT0_TYPE my_chunk[ITEMS_NUM + 1];
+#if IS_DYNAMIC
+    const uint data_set_with_chunk_size = data_set_size + workers_per_data_set;
+    const uint data_set_with_chunk_offset = data_set_idx * data_set_with_chunk_size;
+    const uint my_chunk_offset = data_set_with_chunk_offset + in_data_set_idx * (items_num + 1);
+    __global INPUT0_TYPE* my_chunk = tmp_buffer + my_chunk_offset;
```

Comment:
Can't we fix this global buffer usage something more optimal? 
E.g,, 

If the # of items of each WI is small eough to use stack memory (e.g., 100?) => use stack array 
Otherwise => use global memory. 

I think for bert models the # of items are not that many so we can use stack memory

Reply:
update to use stack array instead of global buffer if `my_chunk_size` is small enough to use stack memory.